joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp

```diff
@@ -302,8 +315,14 @@ class JointTrajectoryController : public controller_interface::ControllerInterfa
   void resize_joint_trajectory_point_command(
     trajectory_msgs::msg::JointTrajectoryPoint & point, size_t size);
 
+  bool set_scaling_factor(const double scaling_factor);
+
   urdf::Model model_;
 
+  realtime_tools::RealtimeBuffer<double> scaling_factor_rt_buff_;
```

Comment:
It would be better to use an std::atomic<double>. The RTBuffer uses a mutex which introduces unnecessary overhead.

Reply:
Will using an std::atomic still allow a guaranteed non-blocking read from RT / a priority for the RT thread? I think we want to avoid having the subscriber callback block the `update()` method.

Reply:
std::atomic is the most lowlevel thread synchronisation type that you can use in c++. In this usecase you just want atomicity (aka. the guarantee that all bytes of the double are updated at once - see https://en.cppreference.com/w/cpp/atomic/memory_order ). 

From my point of view std::atomic will be as less blocking as it can be. (Btw. if you would just use a normal double the access would still be atomic on x86 if the memory is aligned  )

Reply:
I've updated the PR accordingly. This also removes duplication of a shared buffer and a rt-only variable.